crates/runtime/src/datafusion/extension/bytes_processed.rs

```diff
@@ -248,6 +248,14 @@ impl ExecutionPlan for BytesProcessedExec {
         vec![&self.input_exec]
     }
 
+    /// Enforces a 1-1 relationship between `BytesProcessedExec` and its input,
+    /// preventing the introduction of `RepartitionExec` and ensuring that the input
+    /// is not split into sub-inputs. This guarantees that the input is processed
+    /// as a single stream, preserving the order of the data.
+    fn required_input_distribution(&self) -> Vec<Distribution> {
```

Comment:
This should NOT change physical plan until PR is merged. Once it is done snapshots will needs to be updated as there won't be `RepartitionExec` anymore
https://github.com/spiceai/datafusion-federation/pull/20